src/liballoc/boxed.rs

```diff
@@ -749,6 +749,9 @@ impl<T: ?Sized> AsMut<T> for Box<T> {
     }
 }
 
+#[unstable(feature = "pin", issue = "49150")]
+impl<T: ?Sized> Unpin for Box<T> { }
```

Comment:
I tried looking through the issue, and it wasn't immediately clear: why is a Box unpin? After using a pin, we could move out of the box afterwards, right?

Reply:
Best summary is [this comment](https://github.com/rust-lang/rust/issues/49150#issuecomment-409939585) under "adding `Unpin` implementations" TL;DR is you're right, but putting the *box* in the pin doesn't pin the stuff in the heap, only the box's stack representation (i.e. the actual pointer)

Reply:
Both interpretations of Box are valid, but this one is the most useful and consistent with other types (in that pin-projection is related to whether or not the data is stored out-of-line). If a Box is pinned, then there isn't a way to move the underlying data unless we choose to make it `Unpin`.